### PR TITLE
Add plan to do pin tip centring then xray centring

### LINKED ISSUE #499: Full centring algorithm
To do a full centring Hyperion will need to:
* Move to zoom one (applying flat field)
* Move to tip (with offset) at 0
* Move to tip (with offset) at 90
* Grid detect at 90
* Grid detect at 0
* Xray centre

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #499

Note for setting the zoom I just changed the OAVCentring.json file. This will set all OAV parameters twice but to the same values, check on the beamline if this causes much slowness

### To test:
1. Confirm new tests pass

### CODE REVIEW

**Comment by DominicOram** [comment]
Issue seen on beamline where this arms the eiger twice

**Comment by DominicOram** [comment]
Also had issue where we weren't sleeping after setting up the oav params

File: src/artemis/experiment_plans/pin_tip_centring_plan.py [hunk 1]
```diff
@@ -116,6 +116,10 @@ def offset_and_move(tip: Pixel):
 
     LOGGER.info(f"Tip offset in pixels: {tip_offset_px}")
 
+    # need to wait for the OAV image to update
+    # See #673 for improvements
+    yield from bps.sleep(0.3)
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/experiment_plans/pin_tip_centring_plan.py:121]: Yh, it's an annoying factor that needs more thought, have linked your comment to that issue

File: src/artemis/parameters/plan_specific/grid_scan_with_edge_detect_params.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ class GridScanWithEdgeDetectParams(AbstractExperimentParameterBase):
     omega_start: float
 
     # This is the correct grid size for single pin
-    grid_width_microns: Optional[float] = 161
+    grid_width_microns: Optional[float] = 600
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/parameters/plan_specific/grid_scan_with_edge_detect_params.py:35]: It's more sample dependent than beamline. I think https://github.com/DiamondLightSource/python-artemis/issues/860 will cover it

File: src/artemis/experiment_plans/pin_centre_then_xray_centre_plan.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    oav_config_files["oav_config_json"] = parameters.experiment_params.oav_centring_file
+
+    yield from pin_tip_centre_plan(
+        parameters.experiment_params.tip_offset_microns, oav_config_files
+    )
+    grid_detect_params = create_parameters_for_grid_detection(parameters)
+
+    backlight = i03.backlight()
+    aperture_scattergaurd = i03.aperture_scatterguard()
+    detector_motion = i03.detector_motion()
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/experiment_plans/pin_centre_then_xray_centre_plan.py:62]: This should hopefully fall out of the `blueapi` work. We can discuss

File: src/artemis/experiment_plans/pin_centre_then_xray_centre_plan.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        backlight,
+        aperture_scattergaurd,
+        detector_motion,
+        oav_params,
+    )
+
+
+def get_plan(
+    parameters: PinCentreThenXrayCentreInternalParameters,
+) -> Callable:
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/experiment_plans/pin_centre_then_xray_centre_plan.py:76]: General alias will be fixed by `blueapi`. Fixed this return type though.

**Review by Tom-Willemsen — APPROVED** [review]
Happy that my requested changes have been addressed.

> check on the beamline if this causes much slowness

Has this been done? Happy for you to do before or after merging as appropriate...

**Comment by DominicOram** [comment]
> Has this been done? Happy for you to do before or after merging as appropriate...

Yh, this has been tested on the beamline and works without being too slow. May be avenues to speed up but probably not required at this stage

**Review by dperl-dls — APPROVED** [review]
looks good, no additional comments over Tom's review. I like the names of the plans.

**Pull request merged**